### PR TITLE
vscode-extensions: 2021-10-02

### PR DESCRIPTION
**Pull request by eadwu**
###### Motivation for this change
No homework this weekend ... for now ... just the squashed changes I've had in my local tree for god knows how long.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
Thank you!

**Comment by happysalada** [comment]
the diff looks good, it looks like there is a failure for ms-python though ?
https://github.com/NixOS/nixpkgs/pull/140326/checks?check_run_id=3778745669

File: pkgs/misc/vscode-extensions/python/default.nix [hunk 1]
```diff
@@ -41,13 +41,13 @@ in vscode-utils.buildVscodeMarketplaceExtension rec {
...
-    version = "2021.5.829140558";
+    version = "2021.9.1246542782";
   };
 
   vsix = fetchurl {
     name = "${mktplcRef.publisher}-${mktplcRef.name}.zip";
     url = "https://github.com/microsoft/vscode-python/releases/download/${mktplcRef.version}/ms-python-release.vsix";
-    sha256 = "0y2HN4WGYUUXBfqp8Xb4oaA0hbLZmE3kDUXMBAOjvPQ=";
+    sha256 = "sha256-AkwvslTEuqMDBcjReUtcAi1bnobH0SajmWwlcoCQu4A=";
   };
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/python/default.nix:52]: This needs

```
patchelf --replace-needed liblttng-ust.so.0 liblttng-ust.so $out/lib/libcoreclrtraceptprovider.so
``` 
in postPatch

Comment (jraygauthier) [reply pkgs/misc/vscode-extensions/python/default.nix:52]: Sounds like a recipe to fix 137314.

**Comment by jraygauthier** [comment]
Looks like a proper version update for both `vscode.cpptools` and  `ms-python.python`. I am however a bit surprised `ms-python.python` works with latest nixpkgs given recently reported issues #139813 and #137314 which I had no opportunity to look into yet.

**Comment by eadwu** [comment]
Might be a nonessential part of the language server. I didn't come across this error due to other commits in my local nixpkgs ... at least the extension hasn't crashed in the times I've been programming thing using PyTorch.

```diff
diff --git a/pkgs/misc/vscode-extensions/vscode-utils.nix b/pkgs/misc/vscode-extensions/vscode-utils.nix
index 1de3bce3d0d..70c0ebf4005 100644
--- a/pkgs/misc/vscode-extensions/vscode-utils.nix
+++ b/pkgs/misc/vscode-extensions/vscode-utils.nix
@@ -1,4 +1,5 @@
-{ stdenv, lib, buildEnv, writeShellScriptBin, fetchurl, vscode, unzip, jq }:
+{ stdenv, lib, buildEnv, writeShellScriptBin, fetchurl, vscode, unzip, jq
+, autoPatchelfHook, ncurses, zlib }:
 let
   buildVscodeExtension = a@{
     name,
@@ -11,6 +12,7 @@ let
     dontPatchELF ? true,
     dontStrip ? true,
     buildInputs ? [],
+    nativeBuildInputs ? [],
     ...
   }:
   stdenv.mkDerivation ((removeAttrs a [ "vscodeExtUniqueId" ]) // {
@@ -23,6 +25,8 @@ let
     installPrefix = "share/vscode/extensions/${vscodeExtUniqueId}";

     buildInputs = [ unzip ] ++ buildInputs;
+    nativeBuildInputs = [ autoPatchelfHook stdenv.cc.cc ncurses zlib ] ++ nativeBuildInputs;
+    autoPatchelfIgnoreMissingDeps = true;

     installPhase = ''

```

**Review by jraygauthier — APPROVED** [review]
LGTM for both `vscode.cpptools` and `ms-python.python`. I will have to take the time to look into  #139813 and #137314 separately.

File: pkgs/misc/vscode-extensions/wakatime/default.nix [hunk 2]
```diff
@@ -8,23 +8,15 @@ in
...
       publisher = "WakaTime";
-      version = "4.0.9";
-      sha256 = "0sm2fr9zbk1759r52dpnz9r7xbvxladlpinlf2i0hyaa06bhp3b1";
+      version = "17.0.8";
+      sha256 = "0bwnwnmkiv298d1fbd6i20c2k1s8lymhzmawva4zn6dckimllkd8";
     };
 
-    postPatch = ''
-      mkdir wakatime-cli
-      ln -s ${wakatime}/bin/wakatime ./wakatime-cli/wakatime-cli
```
Comment (jraygauthier) [inline pkgs/misc/vscode-extensions/wakatime/default.nix:17]: `wakatime` no longer a dependency in new version but still taken as a parameter.

Comment (jraygauthier) [reply pkgs/misc/vscode-extensions/wakatime/default.nix:17]: I would assume that the extension won't be functional as long as the `wakeatime` is not in `PATH`, right? I might be a good idea to provide nixpkgs's wakeatime as the default `wakeatime` exe if possible when none is explicitly specified via settings.

**Review by jraygauthier — APPROVED** [review]
I just tested that the python extension still work fine. Had a bit of trouble with the yet again changed way of doing thing for pytest test discovery but finally found out what was missing.

I am not working on any c++ projects these times so it as been quite a while since I used the extension. Changes LGTM, I will assume this works fine as well.

**Pull request merged**